src/components/Invitations/InviteProvider.tsx

```diff
@@ -0,0 +1,97 @@
+import React, { createContext, useEffect, useState } from 'react';
+import { Alert } from 'react-native';
+import { t } from 'lib/i18n';
+import { inviteNotifier } from './InviteNotifier';
+import { useAcceptInviteMutation } from '../../hooks/useAcceptInviteMutation';
+import { useAuth } from '../../hooks/useAuth';
+import { useActiveAccount } from '../../hooks/useActiveAccount';
+
+type InviteParams = {
+  inviteId?: string;
+  evc?: string;
+};
+
+type InviteState = {
+  inviteParams: InviteParams;
+  clearPendingInvite: () => Promise<void>;
+};
+
+export const InviteContext = createContext<InviteState>({
+  inviteParams: {},
+  clearPendingInvite: async () => Promise.reject(),
+});
+
+type ProviderProps = {
+  children: React.ReactNode;
+};
+
+export const InviteProvider = ({ children }: ProviderProps) => {
+  const [inviteParams, setInviteParams] = useState<InviteParams>({});
+  const { mutateAsync } = useAcceptInviteMutation();
+  const { authResult, refreshForInviteAccept } = useAuth();
+  const { refetch: refetchAccounts, setActiveAccountId } = useActiveAccount();
+
+  const clearPendingInvite = async () => {
+    setInviteParams({});
+    inviteNotifier.clearCache();
+  };
+
+  // Accept invite if needed
+  useEffect(() => {
+    if (!inviteParams.inviteId || !authResult?.accessToken) {
+      return;
+    }
+
+    // An inviteId is in memory, and the user has authenticated -
+    // time to accept the pending invite.
+    async function acceptInvite() {
+      try {
+        const acceptedInvite = await mutateAsync(inviteParams.inviteId!);
+        await refreshForInviteAccept();
+        await refetchAccounts();
+        setActiveAccountId(acceptedInvite.account);
+      } catch (error) {
+        if (
+          (error as any)?.response?.data?.code === 'INVITATION_ALREADY_ACCEPTED'
+        ) {
+          console.warn('Ignoring already accepted invite');
+        } else {
+          console.warn('Error accepting invitation', error);
+          Alert.alert(t('Error accepting invitation. Please try again.'));
+        }
+      } finally {
+        clearPendingInvite();
+      }
+    }
+    acceptInvite();
+  }, [
+    inviteParams.inviteId,
+    authResult?.accessToken,
+    mutateAsync,
+    refetchAccounts,
+    refreshForInviteAccept,
+    setActiveAccountId,
+  ]);
+
+  // Listen to pending invite param notifications
+  useEffect(() => {
+    const listener = (pendingInviteParams: InviteParams) => {
+      setInviteParams(pendingInviteParams);
+    };
+    inviteNotifier.addListener(listener);
```

Comment:
Very cool, so an external link provider can emit invite events via the `inviteNotifier` and then the `InviteProvider` can handle the invite acceptance 👍